### ISSUE TITLE
Using certain words (slurs) should result in being automatically kicked and/or banned.

### ISSUE THREAD

**Issue by KrulPax** [ScorpioStation/ScorpioStation#319]
The idea behind this is that we already had a bunch of uhm, _not very pleasant individuals_ that decided to vent their frustrations about something in a terrible fashion. I'm talking about idiots spewing the n-word over comms when something displeases them. While as far as I understand the administrative team would prefer to find out and attempt to correct any wrongings that may be happening, and I am 100% for this in most cases, I believe that if someone resolves to use vocabulary like that they already strip themselves of any chance for understanding from me at least, given that they at that point attempt to be disruptive and provocative.
With that reasoning, I'd like to propose an automated system that would automatically block the use of certain slurs while also automatically kicking the player that resolved to use the said word, and additionally sending a warning to the administrative team about the situation at hand. If this idea were to be considered, I'd like to leave a bunch of bullet points as to what should be taken into account when designing such a system, in my personal opinion:

- The system should take into account usage of in-game PDA messages, paper + pen communications, name tags using the hand labeler, IDs and other modifiable tools of identification, station-wide announcements, use of radio, and of course the classic face to face communications, since people that like to engage in disruptive behavior like this are often extremely creative in their attempts at being a terrible human being.
- To avoid the confusion of the rest of the crew when someone suddenly passes out after being automatically kicked, they should get an automated OOC information that a certain player has been kicked for their attempt at using the slur. 

I believe that this would successfully depower the idiots whose sole goal is to get a kick out of someone being angry at them. It would also cut short a lot of unpleasant situations that unfortunately do happen on the Scorpio Server. Lastly, I believe that would save a lot of time for the administrative team - there were instances of situations where the ban for slurs like this was given a handful of minutes after the situation since there are times when admins are simply not on the server, and that's, of course, okay, but without a system like this, it leads to forcing a swift resolution of the problem, which probably is not very pleasant to the administrative team.

**Comment by TheMaveran** [comment]
No.. I think automated systems like this are a bad idea no matter the intent. I can only think of like two extremely racist words that are bad no matter the context. But for the rest of them, a system like this would just ignore context and issue out false bans. And then there is the issue of who decides the 'bad words'. Not worth it.

**Comment by KrulPax** [comment]
Well I kinda agree, I rather not have some misunderstanding over things like this... but like you said there are at least "two extremely racist words that are bad no matter the context" and that's basically what I refer to here. It's not policing swears for crying out loud, but slurs, and I think I did explain thoughtfully in the first paragraph that I don't mean just any ol' "bad words" as you called them, but the kind of language that people use as a provocation. And while I agree that making such an automated system for a big chunk of words would be problematic, banning for example the ever so popular n-word in such a manner would be only healthy for the community of the server.

**Comment by Kiyahitayika** [comment]
These are questions for the server heads. They should be consulted.

**Comment by KrulPax** [comment]
That is the purpose of this issue entry, yes.

**Comment by blinkdog** [comment]
> No.. I think automated systems like this are a bad idea no matter the intent. I can only think of like two extremely racist words that are bad no matter the context. But for the rest of them, a system like this would just ignore context and issue out false bans. And then there is the issue of who decides the 'bad words'. Not worth it.

There was some work a few years back on using machine learning to detect toxic comments. It's actually pretty good (for a computer), but [definitely not perfect](https://www.technologyreview.com/2017/02/24/153573/its-easy-to-slip-toxic-language-past-alphabets-toxic-comment-detector/).

I do think that @KrulPax is suggesting a very limited system intended for the worst trolls. A simple automated system to ban individuals who drop the N-word? Sure. I'd be in favor of that. It means an admin can focus on fun stuff like events rather than take time out to deal with the troll de jure. But as @TheMaveran points out, to avoid false bans it wouldn't be wise to expand it beyond a few very limited cases.

**Comment by KrulPax** [comment]
I wholeheartedly agree! I hope that at least I am opening a discussion about potentially utilizing such a tool, and maybe it is a way to address the problem at least partially. All and all, like Kiya said this is something for the administrative team to consider.
